cosine_similarity_model.py

```diff
@@ -0,0 +1,179 @@
+import json
+import numpy as np
+from sklearn.metrics.pairwise import cosine_similarity
+from pprint import pprint
+from collections import Counter
+import logging
+
+def setup_logger():
+    logger = logging.getLogger('logfile.log')
+    logger.setLevel(logging.INFO)
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    return logger
+
+
+logger = setup_logger()
+
+
+def load_movies_from_json(file_path):
+    with open(file_path, "r", encoding="utf-8") as file:
+        data = json.load(file)
+    return data
+
+
+def extract_genres(movies):
+    all_genre_ids = set()
+    for movie in movies:
+        details = movie.get('detail', [])
+        for detail in details:
+            genres = detail.get('genres', [])
+            for genre in genres:
+                all_genre_ids.add(genre['id'])
+    return list(all_genre_ids)
+
+
+def extract_actors(movies):
```

Comment:
for cast, use only the top 3 actors, we don't need all the minor roles
this can be done as

```
cast = movies_df['cast'].apply(
    lambda x: x[:3] if len(x) >= 3 else x
)
```